doc/fundamentals/lifecycle_of_a_datum.rst

```diff
@@ -0,0 +1,549 @@
+you can use “union” and “cross” operations to combine inputs.
+
+`Union
+input <http://docs.pachyderm.io/en/latest/reference/pipeline_spec.html#union-input>`__
+will combine each of the datums in the input repos as one set of datums.
+The result is that the number of datums processed is the sum of all the
+datums in each repo.
+
+For example, let’s say you have two input repos, A and B. Each of them
+contain three files with the same names: 1.txt,
+2.txt, and 3.txt. Each file hashes
+differently, because each of the files contains different content,
+despite the identical filenames. If you were to cross them in a
+pipeline, the “input” object in the pipeline spec might look like this.
+
+::
+
+     "input": {
+         "union": [
+             {
+                 "pfs": {
+                     "glob": "/*",
+                     "repo": "A"
+                 }
+             },
+             {
+                 "pfs": {
+                     "glob": "/*",
+                     "repo": "B"
+                 }
+             }
+         ]
+     }
+
+If each repo had those three files at the top, there would be six (6)
+datums overall, which is the sum of the number of input files. You’d see
+the following datums, in a random order, in your pipeline as it ran
+though them.
+
+::
+
+   /pfs/A/1.txt
+
+   /pfs/A/2.txt
+
+   /pfs/A/3.txt
+
+   /pfs/B/1.txt
+
+   /pfs/B/2.txt
+
+   /pfs/B/3.txt
+
+This is shown in the animation below.
+Remember that the order in which the datums will be processed is not guaranteed;
+we'll show them in a random order.
+
+.. image:: A-B-union-no-names.gif
+    :align: center
+    :alt: Each datum will get be visible to the input repo of a pipeline using a cross input
+
+One of the ways you can make your code simpler is to use the ``name``
+field for the ``pfs`` object and give each of those repos the same name.
+
+::
+
+     "input": {
+         "union": [
+             {
+                 "pfs": {
+                     "name": "C",
+                     "glob": "/*",
+                     "repo": "A"
+                 }
+             },
+             {
+                 "pfs": {
+                     "name": "C",
+                     "glob": "/*",
+                     "repo": "B"
+                 }
+             }
+         ]
+     }
+
+You’d still see the same six datums, in a random order, in your pipeline
+as it ran though them, but they’d all be in a directory with the same
+name: C.
+
+::
+
+   /pfs/C/1.txt  # from A
+
+   /pfs/C/2.txt  # from A
+
+   /pfs/C/3.txt  # from A
+
+   /pfs/C/1.txt  # from B
+
+   /pfs/C/2.txt  # from B
+
+   /pfs/C/3.txt  # from B
+
+This is shown in the animation below.
+Remember that the order in which the datums will be processed is not guaranteed,
+so we show them in a random order.
+We also highlight the input file at the top of the image,
+so it’s easier to understand when a file from a particular repo is being processed.
+
+.. image:: A-B-union-names.gif
```

Comment:
the example is good, but the gif is way too confusing. Let's figure out how to simplify. Same for the other union gif.